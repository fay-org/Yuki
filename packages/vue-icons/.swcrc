{
  "jsc": {
    "experimental": {
      "plugins": [["swc-plugin-vue-jsx", {}]]
    }
  }
}
