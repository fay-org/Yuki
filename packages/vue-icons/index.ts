import { build } from './scripts/bundler'

//  iife ~
;(() => build())()
