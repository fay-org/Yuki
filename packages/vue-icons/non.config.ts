import path from 'path'
import { defineNonConfig } from '@fect-ui/cli'

export default defineNonConfig({
  lib: {
    input: path.join(__dirname, 'packages'),
    format: ['cjs', 'es']
  }
})
