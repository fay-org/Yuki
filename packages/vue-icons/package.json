{
  "name": "@fect-ui/vue-icons",
  "version": "2.0.1",
  "author": "XeryYue",
  "description": "Well prepared high quality Icon Library for fect",
  "scripts": {
    "build": "rm -rf dist types && ts-node scripts/bundler.ts"
  },
  "main": "dist/cjs/index.cjs.js",
  "module": "dist/esm/index.esm.js",
  "types": "types/index.d.ts",
  "license": "MIT",
  "files": [
    "LISCENSE",
    "dist",
    "types/",
    "README.md"
  ],
  "devDependencies": {
    "jsdom": "^16.6.0",
    "@types/svgo": "1.3.2",
    "svgo": "1.3.2",
    "@types/jsdom": "^16.2.11",
    "ora": "5.4.1"
  },
  "peerDependencies": {
    "vue": "^3.0.11"
  },
  "sideEffects": false,
  "homepage": "https://github.com/fect-org/fect",
  "repository": {
    "type": "git",
    "url": "https://github.com/fect-org/fect"
  }
}
