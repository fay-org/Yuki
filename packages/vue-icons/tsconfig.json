{
  "compilerOptions": {
    "baseUrl": ".",
    "outDir": "./lib",
    "target": "ES2017",
    "module": "CommonJS",
    "sourceMap": false,
    "declaration": true,
    "skipLibCheck": true,
    "esModuleInterop": true,
    "importHelpers": true,
    "allowJs": true,
    "resolveJsonModule": true,
    "jsx": "preserve",
    "lib": ["esnext", "dom"]
  },
  "include": ["scripts/**/*"]
}
