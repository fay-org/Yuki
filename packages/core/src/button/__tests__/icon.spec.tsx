import { mount } from '@vue/test-utils'
import { Github } from '@fect-ui/vue-icons'
import Button from '..'

describe('ButtonIcon', () => {
  it('icon', () => {
    const wrapper = mount(Button, {
      slots: {
        default: () => 'Action',
        icon: () => <Github />
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
