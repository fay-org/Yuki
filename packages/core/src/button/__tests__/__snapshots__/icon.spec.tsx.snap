// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonIcon icon 1`] = `
"<button class="fect-button fect-button--default" style="--button-height: calc(2.5 * 16px); --button-min-width: calc(10.5 * 16px); --button-radius: 6px; --button-font-size: calc(0.875 * 16px); --button-shadow: 0 5px 10px rgba(0, 0, 0, 0.12); --button-shadow-hover: 0 8px 30px rgba(0, 0, 0, 0.12); --button-pt: 0; --button-pr: calc(1.375 * 16px); --button-pb: 0; --button-pl: calc(1.375 * 16px); --button-mt: 0; --button-mr: 0; --button-mb: 0; --button-ml: 0; --button-bg-color: #fff; --button-border-color: #eaeaea; --button-color: #666; --button-hover-bg-color: #fff; --button-hover-border-color: #000; --button-hover-color: #000;" type="button">
  <!---->
  <!---->
  <div class="fect-button__icon" style="left: 15%; transform: translateY(-50%);"><svg width="24" height="24" fill="none" stroke="currentColor" stroke-linecap="round" stroke-linejoin="round" stroke-width="1.5" shape-rendering="geometricPrecision" style="" viewBox="0 0 24 24">
      <path d="M9 19c-5 1.5-5-2.5-7-3m14 6v-3.87a3.37 3.37 0 0 0-.94-2.61c3.14-.35 6.44-1.54 6.44-7A5.44 5.44 0 0 0 20 4.77 5.07 5.07 0 0 0 19.91 1S18.73.65 16 2.48a13.38 13.38 0 0 0-7 0C6.27.65 5.09 1 5.09 1A5.07 5.07 0 0 0 5 4.77a5.44 5.44 0 0 0-1.5 3.78c0 5.42 3.3 6.61 6.44 7A3.37 3.37 0 0 0 9 18.13V22"></path>
    </svg></div>
  <div class="fect-button__text">Action</div>
</button>"
`;
